lib/std/fmt/parse_float.zig

```diff
@@ -1,377 +1,195 @@
-// Adapted from https://github.com/grzegorz-kraszewski/stringtofloat.
-
-// MIT License
 //
-// Copyright (c) 2016 Grzegorz Kraszewski
+// Adapted from sqlite3's sqlite3AtoF()
+// https://github.com/mackyle/sqlite/blob/ae5d3aa91a794f6f1486b3f453ec44c0aa4c086b/src/util.c#L375-L571
 //
-// Permission is hereby granted, free of charge, to any person obtaining a copy
-// of this software and associated documentation files (the "Software"), to deal
-// in the Software without restriction, including without limitation the rights
-// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-// copies of the Software, and to permit persons to whom the Software is
-// furnished to do so, subject to the following conditions:
+// adds support for parsing f128 floats, "nan", "inf"
 //
-// The above copyright notice and this permission notice shall be included in all
-// copies or substantial portions of the Software.
+// Returns a float if the string is a valid number, or error.invalidCharacter
+// if the string is empty or contains extraneous text.  Valid numbers
+// are in one of these formats:
 //
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-// SOFTWARE.
+//    [+-]digits[E[+-]digits]
+//    [+-]digits.[digits][E[+-]digits]
+//    [+-].digits[E[+-]digits]
 //
-
-// Be aware that this implementation has the following limitations:
+// underscore characters may appear zero or more times between
+// digits and are ignored by the parser: 500_000
 //
-// - Is not round-trip accurate for all values
-// - Only supports round-to-zero
-// - Does not handle denormals
 
 const std = @import("std");
 const ascii = std.ascii;
 
-// The mantissa field in FloatRepr is 64bit wide and holds only 19 digits
-// without overflowing
-const max_digits = 19;
-
-const f64_plus_zero: u64 = 0x0000000000000000;
-const f64_minus_zero: u64 = 0x8000000000000000;
-const f64_plus_infinity: u64 = 0x7FF0000000000000;
-const f64_minus_infinity: u64 = 0xFFF0000000000000;
-
-const Z96 = struct {
-    d0: u32,
-    d1: u32,
-    d2: u32,
-
-    // d = s >> 1
-    inline fn shiftRight1(d: *Z96, s: Z96) void {
-        d.d0 = (s.d0 >> 1) | ((s.d1 & 1) << 31);
-        d.d1 = (s.d1 >> 1) | ((s.d2 & 1) << 31);
-        d.d2 = s.d2 >> 1;
-    }
+fn caseInEql(a: []const u8, b: []const u8) bool {
```

Comment:
this function is available with `ascii.eqlIgnoreCase`

Reply:
done.